pkg/cmd/roachtest/tests/indexbackfiller.go

```diff
@@ -0,0 +1,259 @@
+			t.Fatal(err)
+		}
+
+		t.Status("starting validation")
+		err := validateSecondaryIndex(db, `tpcc."order"`, oldIdx, newIdx)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	// runBackfillInvertedIndex first creates a table with a JSON column and loads
+	// it with 10 minutes of JSON workload. It then creates an inverted index
+	// while the JSON workload runs. After the index creation and the workload
+	// finishes, the test creates another index on the same columns using the old
+	// index backfiller, and compares the fingerprints of the two indexes.
+	runBackfillInvertedIndex := func(ctx context.Context, t test.Test, c cluster.Cluster) {
+		roachNodes, loadNode := setupTest(ctx, t, c)
+
+		cmd := fmt.Sprintf("./workload init json {pgurl%s}", roachNodes)
+		if err := c.RunE(ctx, loadNode, cmd); err != nil {
+			t.Fatal(err)
+		}
+
+		db := c.Conn(ctx, t.L(), roachNodes[0])
+		defer db.Close()
+
+		initialDataDuration := time.Minute * 10
+		indexDuration := 2 * initialDataDuration
+
+		// First generate random JSON data using the JSON workload.
+		m := c.NewMonitor(ctx, roachNodes)
+
+		cmdWrite := fmt.Sprintf(
+			"./workload run json --read-percent=0 --duration %s {pgurl%s} --batch 1000 --sequential",
+			initialDataDuration.String(), roachNodes,
+		)
+		m.Go(func(ctx context.Context) error {
+			c.Run(ctx, loadNode, cmdWrite)
+
+			db := c.Conn(ctx, t.L(), 1)
+			defer db.Close()
+
+			var count int
+			if err := db.QueryRow(`SELECT count(*) FROM json.j`).Scan(&count); err != nil {
+				t.Fatal(err)
+			}
+			t.L().Printf("finished writing %d rows to table", count)
+
+			return nil
+		})
+
+		m.Wait()
+
+		// Run the workload (with both reads and writes), and create the index at the same time.
+		m = c.NewMonitor(ctx, roachNodes)
+		oldIdx := "idx_old_ib"
+		newIdx := "idx_new_ib"
+		createFmt := `CREATE INVERTED INDEX %s ON json.j (v)`
+
+		cmdWriteAndRead := fmt.Sprintf(
+			"./workload run json --read-percent=50 --duration %s {pgurl:1-%d} --sequential",
+			indexDuration.String(), c.Spec().NodeCount-1,
+		)
+		m.Go(func(ctx context.Context) error {
+			c.Run(ctx, loadNode, cmdWriteAndRead)
+			return nil
+		})
+
+		m.Go(func(ctx context.Context) error {
+			db := c.Conn(ctx, t.L(), 1)
+			defer db.Close()
+
+			t.L().Printf("creating index %s", newIdx)
+			start := timeutil.Now()
+			if _, err := db.Exec(fmt.Sprintf(createFmt, newIdx)); err != nil {
+				return err
+			}
+			t.L().Printf("index was created, took %v", timeutil.Since(start))
+
+			return nil
+		})
+
+		m.Wait()
+
+		createStmt := fmt.Sprintf(createFmt, oldIdx)
+		alterCmd := `SET CLUSTER SETTING sql.mvcc_compliant_index_creation.enabled = false; %s;`
+
+		t.Status("creating index using old index backfiller")
+		if _, err := db.ExecContext(ctx, fmt.Sprintf(alterCmd, createStmt)); err != nil {
+			t.Fatal(err)
+		}
+
+		t.Status("starting validation")
+		err := validateSecondaryIndex(db, "json.j", oldIdx, newIdx)
```

Comment:
I don't think that EXPERIMENTAL_FINGERPRINT works on inverted indexes.

Reply:
Thanks!  I see that it fails now for inverted indexes. It also seems to fail with partial indexes as well with this error:
```
hash-fingerprint: index "idx_new_ib" is a partial index that does not contain all the rows needed to execute this query
```

I've kept the statements for regular, expression, and unique indexes in the test.

Reply:
Thanks! Just did some quick verification that EXPERIMENTAL_FINGERPRINT doesn't work for inverted, expression, and partial indexes. I just added some more regular and unique index creation test cases for a different table.

Reply:
Awesome, thanks!  Doesn't need to block this PR, but I took a peak at support partial indexes and it seems like that was relatively simple so I added that here: https://github.com/cockroachdb/cockroach/pull/80539